### ISSUE TITLE
Scan `LICENSES` folder (and maybe other common folders)

### ISSUE THREAD

**Issue by AFCMS** [licensee/licensee#737]
### Is your feature request related to a problem? Please describe the problem you're trying to solve.

If you want to have multiple license files in your project, you may want to add them inside a subfolder.

The [REUSE](https://reuse.software) project which aims to make licensing more strict require the licenses files to be in a `LICENSES` folder in the project root folder.

I would like to follow REUSE recommendations and be able to run the associated linter, but if I put my licenses in a folder they are no longer recognized and displayed by GitHub which kinda defeat the point of having clear licensing.

### Describe the solution you'd like

Scan not only the root folder but also the `LICENSES` folder for licenses (and maybe other common folder names?).

Would be solved by https://github.com/licensee/licensee/issues/603, but this would have a bigger performance hit I guess.

### Describe alternatives you've considered

None

### Additional context

None

**Comment by AliSajid** [comment]
I came here to create this issue, but I'll just +1 this. I would like this targeted functionality because it will allow REUSE compliance and GitHub previews at the same time. Happy to contribute a PR if needed.

**Comment by vladh** [comment]
@mlinksva I am happy to implement REUSE support. Would this be welcome?

**Comment by yuhr** [comment]
For projects with multiple `LICENSES/*.txt`, it would be somewhat complicated how to determine which license to show in the side column of the front page of a repository. In the meantime, at least symlinking `LICENSE` to `LICENSES/<main-license-that-covers-core-part-of-your-project>.txt` (which doesn't work for now) should be supported first.

**Comment by vladh** [comment]
On Sourcehut, we show all licenses a project uses. If there are too many to show, we truncate them: “Apache-2.0, GPL-3.0-only, and 3 more”.

**Comment by vladh** [comment]
@TristanWasTaken I agree, we can find a solution along those lines.

@mlinksva Trying to check in again — I can implement this, would this be welcome?

**Comment by yuhr** [comment]
> why not just show multiple names if it's only a few, or "x licenses found" if it's too many?

That sounds at least okay to me, but sorting in some set order or by popularity would not be great for UX. I imagined more semantic way such as scanning every file for the `SPDX-License-Identifier` headers and sorting by occurrence, as well as “sectioning” licenses like this:

- Source Code: `MPL-2.0`
- Examples: `CC0-1.0`
- Documentation: `CC0-1.0`
- Assets: `CC-BY-3.0` and 2 more licenses
- Configs: `CC0-1.0`

**Comment by vladh** [comment]
> That sounds at least okay to me, but sorting in some set order or by popularity would not be great for UX. I imagined more semantic way such as scanning every file for the `SPDX-License-Identifier` headers and sorting by occurrence, as well as “sectioning” licenses like this:

I don't think this is our problem to solve. What `licensee` can implement, which I am offering to do, is to use the [REUSE specification][reuse] to return information on all licenses used in a compliant project, as implemented by [reuse-tool] and [SourceHut][srht]. The _presentation_ of these licenses, so how many licenses are displayed, which licenses are displayed, how they are grouped, and so on, should be decided by those _using_ `licensee` — for example, the GitHub UI team — not by us.

[reuse]: https://reuse.software/spec-3.2/
[reuse-tool]: https://github.com/fsfe/reuse-tool
[srht]: https://git.sr.ht/~vladh/fotografiska/licenses

**Comment by NamesCode** [comment]
Had the same issue, bumping this ^^

**Comment by sdruskat** [comment]
+1 to implement this as described in https://github.com/licensee/licensee/issues/737#issuecomment-2473783478.

**Comment by erickt** [comment]
Note that I have a PR up that provides a simplified version of this in #820 that just searches the `LICENSES/` directory. It's not as full featured as directly using `reuse-tool` as suggested by @vladh in https://github.com/licensee/licensee/issues/737#issuecomment-2473783478, but doesn't have to solve the distribution challenge. As far as I can tell there isn't a ruby library that implements the REUSE spec, so calling it would require all clients to also setup a python environment so we could shell out to it.
